### PR TITLE
Reunification

### PR DESCRIPTION
**Pull request by jaredhirsch**
@nchapman @meandavejustice @pdehaan r?

This PR cleans up the build chain and gets FxA handling working with ampersand.

I've separated the changes into two commits. The first commit just copies over the contents of the idea-town-app repo. The new code is in the second commit; that's the stuff that needs a review.

Build chain:
- base template has been copied over from old `idea-town-app` repo
- gulp converts source code in `static-src` to django-visible built stuff in `static`
- django inserts style/script links into the base template

New FxA / session lifecycle:
- Front-end redirects to django-allauth /login endpoint to start flow
- django-allauth redirects back to '/' after FxA login succeeds
- Logout is handled by front-end via xhr POST to /logout

Note that we are not using the fxa-relier-client repo; instead the lifecycle revolves around the /login and /logout endpoints exposed by django-allauth.

There are a lot of endpoints still templated by django, but this PR has gotten big enough and taken long enough. I'll file followup bugs to capture remaining reunification tasks.

Major todo introduced by this code (I'm open to fixing this before landing it):
- We need to create an api endpoint to send session info to the front-end
  - Currently, we've disabled CSP and httponly session cookies to get the roundtrip working

### CODE REVIEW

File: idea_town/frontend/static-src/styles/vendor/mozilla-tabzilla/css/tabzilla.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#tabzilla {
```
Comment (meandavejustice) [inline idea_town/frontend/static-src/styles/vendor/mozilla-tabzilla/css/tabzilla.scss]: Should this and the normalize dep be checked in here? or would we rather have a gulp task which copies them out of `node_modules`?

Comment (jaredhirsch) [inline idea_town/frontend/static-src/styles/vendor/mozilla-tabzilla/css/tabzilla.scss]: Ah, you're right - there is such a gulp task, but the .gitignore is slightly incorrect. I'll fix

**Comment by pdehaan** [comment]
@6a68 I'm getting a couple 404s on the /static/ assets, and an odd error:

> **TypeError: undefined has no properties** (in app.js:515:5)

Which seems to be roughly this code:

``` js
  template: function template(ctx) {
    return _mustache2['default'].render(undefined._template, ctx);
  },
```

Which looks dangerously close to this source ES6:

``` js
template: (ctx) => { return mustache.render(this._template, ctx); },
```

&mdash; via https://github.com/mozilla/idea-town/pull/59/files#diff-56c208341a408142c3b040a231e2664eR17

idea_town

idea_town

**Comment by pdehaan** [comment]
Another curiosity is that /mozilla-logo.png file. It shows up as a broken image, yet DevTools says its a HTTP 200:

```
server_1       | [10/Sep/2015 22:44:59] "GET /mozilla-logo.png HTTP/1.1" 200 1520
```

Is it because there isn't any mozilla-logo.png in https://github.com/mozilla/idea-town/tree/master/idea_town/frontend/static-src/images?

Is there a problem with our 404 route? http://ideatown.dev:8000/mozilla-logo.png seems to show the index route and not a broken image icon or 404 page or whimsical GIF.

I'm starting to think its a 404 issue. I can go to http://ideatown.dev:8000/__version__ and it displays the home route instead of a 404 page or JSON blob.

**Comment by pdehaan** [comment]
HTML linting looks good. The [W3C validator](https://validator.w3.org/#validate_by_input+with_options) only complains about the lack of `src` and `alt` attributes on a couple `` tags:
showing_results_for_contents_of_text-input_area_-_nu_html_checker

**Comment by meandavejustice** [comment]
```
server_1       | [10/Sep/2015 23:15:31] "GET / HTTP/1.1" 200 1501
server_1       | [10/Sep/2015 23:15:31] "GET /mozilla-logo.png HTTP/1.1" 200 1501
server_1       | [10/Sep/2015 23:15:31] "GET /static/styles/vendor/normalize.css/normalize.css HTTP/1.1" 404 1845
server_1       | [10/Sep/2015 23:15:31] "GET /static/styles/vendor/mozilla-tabzilla/css/tabzilla.css HTTP/1.1" 404 1863
server_1       | [10/Sep/2015 23:15:31] "GET /static/styles/main.css HTTP/1.1" 200 19771
server_1       | [10/Sep/2015 23:15:31] "GET /static/app/app.js HTTP/1.1" 200 524032
```

**Comment by pdehaan** [comment]
So, after tweaking the `template()` method and getting it working locally, everything seems to look great!

I did notice that clicking on the experiment name links in the boilerplate experiments doesn't seem to do anything though.

idea_town_and_inbox__450__-_pdehaan_mozilla_com_-_mozilla_mail

File: idea_town/frontend/templates/idea_town/frontend/index.html [hunk 2]
```diff
@@ -1,13 +1,49 @@
...
+      <h1>Idea Town is the Best</h1>
+    </div>
+    <div id="experiments"></div>
+  </div>
+  <footer>
+    <ul>
+      <li><a href="//mozilla.com"><img src="mozilla-logo.png"></a>
+      <li><a href="//blog.mozilla.org">Blog</a>
+      <li><a href="//mozilla.org/privacy">Privacy Policy</a>
+      <li><a href="//mozilla.org/cookies">Cookies</a>
```
Comment (pdehaan) [inline idea_town/frontend/templates/idea_town/frontend/index.html:27]: Ref: #67

I think this link should be `//mozilla.org/privacy/websites/#cookies` (per the [**idea-town-splash**](https://github.com/mozilla/idea-town-splash/blob/6bb0ca832896249240c76fb745ef143c1d446387/routes/data.json#L43-L46) repo).

**Comment by jaredhirsch** [comment]
@pdehaan yeah, that's just placeholder junk we'll be removing when the for reals UX is ready

File: idea_town/frontend/templates/idea_town/frontend/index.html (hunk 2 above)
Comment (jaredhirsch) [inline idea_town/frontend/templates/idea_town/frontend/index.html:27]: I've opened a followup bug to finalize the list of links when we get final UX (#68).

**Comment by jaredhirsch** [comment]
@pdehaan The 404 route isn't currently working; we should fix this as part of the overall bug to remove templated routes still existing in django. I'll update #65 to add this task.

**Comment by jaredhirsch** [comment]
@meandavejustice Is this thing ready to merge? What do you think?

**Comment by meandavejustice** [comment]
:+1:

**Pull request merged**